text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,430 @@
+> privacy policy enforcement can be implemented at various levels (e.g., in the
+> generated client SDK or in a proxy).
+>
+> Note 5: This
+> recent [paper](https://arxiv.org/pdf/2311.07509.pdf#:~:text=The%20results%20of%20the%20benchmark%20provide%20evidence%20that%20supports%20our,LLM%20without%20a%20Knowledge%20Graph)
+> from [data.world](https://data.world/home/), along with
+> the [MetricFlow framework](https://docs.getdbt.com/docs/build/about-metricflow)
+> which underpins the [dbt Semantic Layer](https://www.getdbt.com/product/semantic-layer),
+> highlights the significance of adopting a schema-first approach in data
+> modeling, especially for Generative AI-based question answering systems. Tools
+> like Observability Query Assistants (
+> e.g. [Elastic AI Assistant](https://www.elastic.co/fr/blog/introducing-elastic-ai-assistant)
+> and [Honeycomb Query Assistant](https://www.honeycomb.io/blog/introducing-query-assistant?utm_source=newswire&utm_medium=link&utm_campaign=query_assistant))
+> are likely to become increasingly prevalent and efficient in the near future,
+> thanks to the adoption of a schema-first approach.
+
+Several OTEPs will be dedicated to the precise definition of the structure and
+the format of this/these schema(s). The rules for resolving overrides
+(inheritance), external references, and conflicts will also be described in
+these OTEPs. See the Roadmap section for a comprehensive list of these OTEPs.
+
+## Resolved Telemetry Schema
+
+A Resolved Telemetry Schema is the outcome of the schema resolution process.
+This process involves taking the entire hierarchy of Telemetry Schemas and
+Semantic Convention Registries and applying a set of rules to resolve overrides
+and eliminate external references. The key design principles to be followed in
+the definition of the Resolved Telemetry Schema are:
+
+* **Self-contained**: No external references are allowed. This artifact contains
+  everything required to determine what an application or a library produces in
+  terms of telemetry.
+* **Easy to exchange**: This artifact must be easily accessible from a web
+  server via a URL. This artifact must be small and avoid the repetition of
+  definitions.
+* **Easy to parse**: A widespread and well-defined format should be preferred.
+  JSON is an example of such a format.
+* **Easy to interpret**: The internal structure of this artifact must be
+  straightforward to avoid any misinterpretation.
+* **Platform- and Language-agnostic**: This artifact must be independent of any
+  platform architectures and programming languages.
+
+The following diagram describes two main use cases for the Resolved Telemetry
+Schema. The key points to remember are: 1) both use cases result in a Resolved
+Telemetry Schema, and 2) Resolved Telemetry Schemas serve as the mechanism for
+distributing Telemetry Schemas throughout the entire ecosystem.
+
+![Use cases](./img/0240-otel-weaver-use-cases.svg)
```

Comment:
Is it required for end-users to use a Component Telemetry Schema? Or could they just use the Resolved Telemetry Schema provided by OTel, assuming that no customizations are needed?

Reply:
Yes it is conceivable to use a resolved telemetry schema, representing a compact and resolved version of a semantic convention registry, in a manner similar to that of current SDK/API clients, where nothing specific is specified/known.
However, the preferred mode (or main purpose) of use for the component telemetry schema in this proposal is to define the telemetry objects produced by an application or a library.